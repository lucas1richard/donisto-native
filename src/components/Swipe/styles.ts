import {
  StyleSheet,
  Dimensions
} from 'react-native';
import theme from 'theme/iftheme';

export const winWidth = Dimensions.get('window').width;
const imgWidth = winWidth;

const pageDot = {
  backgroundColor: theme.white,
  borderWidth: 1,
  borderColor: theme.primary[400],
  width: 12,
  height: 12,
  borderRadius: 6,
  marginHorizontal: 4
};

const styles = StyleSheet.create({
  image: {
    height: imgWidth,
    width: imgWidth
  },
  pageDot,
  pageDotActive: {
    ...pageDot,
    backgroundColor: theme.primary[400]
  },
  dotSeparater: {
    width: 4
  },
  dotWrapper: {
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: 4
  },
  wrapper: {
    flex: 1,
    position: 'relative',
    marginBottom: 12
  },
  imgWrapper: {
    width: winWidth
  }
});

export default styles;
