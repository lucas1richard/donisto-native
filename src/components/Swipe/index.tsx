import * as React from 'react';
import { View, ScrollView } from 'react-native';
import styles from './styles';

export interface ISwipeProps {
  height?: number;
  hideIndex?: boolean;
  onIndexChanged?(): any;
}

class Swipe extends React.PureComponent<ISwipeProps> {
  state = {
    page: 0
  };

  onScroll = (ev: any) => {
    const { nativeEvent } = ev;
    const {
      layoutMeasurement,
      contentOffset
    } = nativeEvent;
    const page = Math.round(contentOffset.x / layoutMeasurement.width);
    this.setState({
      page
    });
    if (typeof this.props.onIndexChanged === 'function') {
      this.props.onIndexChanged();
    }
  }

  render() {
    const { page } = this.state;
    const { children, height, hideIndex } = this.props;

    const wrapperStyle: any[] = [styles.imgWrapper];
    if (height) {
      wrapperStyle.push({ height });
    }

    return (
      <View style={styles.wrapper}>
        <ScrollView
          horizontal
          pagingEnabled
          showsHorizontalScrollIndicator={false}
          scrollEventThrottle={1000000}
          onScroll={this.onScroll}
        >
          {React.Children.map(children, (child) => child && (
            <View style={wrapperStyle}>
              {child}
            </View>
          ))}
        </ScrollView>
        {!hideIndex && (
          <View style={styles.dotWrapper}>
            {React.Children.map(children, (child, ix) => child && (
              <View
                style={page === ix ? styles.pageDotActive : styles.pageDot}
              />
            ))}
          </View>
        )}
      </View>
    );
  }
}

export default Swipe;
