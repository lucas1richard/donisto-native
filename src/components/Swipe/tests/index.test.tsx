import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Swipe from '..';

describe('<Swipe />', () => {
  const renderComponent = () => toJson(shallow(<Swipe />));
  it('renders', () => {
    expect(renderComponent()).toMatchSnapshot();
  });
});