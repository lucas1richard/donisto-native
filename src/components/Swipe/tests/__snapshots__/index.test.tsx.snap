// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Swipe /> renders 1`] = `
<Component
  style={
    Object {
      "flex": 1,
      "marginBottom": 12,
      "position": "relative",
    }
  }
>
  <ScrollViewMock
    horizontal={true}
    onScroll={[Function]}
    pagingEnabled={true}
    scrollEventThrottle={1000000}
    showsHorizontalScrollIndicator={false}
  />
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 4,
      }
    }
  />
</Component>
`;
